### PR TITLE
[msbuild] Make the default value for IsMacEnabled property dependent upon the OS.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Also make it overridable by only setting it if it's not already set.

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-9892/1/display/redirect) (on internal Jenkins) succeeded

✅ [Provisioning succeeded](https://jenkins.internalx.com/job/macios/job/PR-9892/1//console)
✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-9892/1//console)
✅ Packages built successfully
<details><summary>View packages</summary>

* [xamarin.ios-14.7.0.94.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/xamarin.ios-14.7.0.94.pkg)
* [Microsoft.iOS.Ref.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.iOS.Ref.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.iOS.Runtime.ios-arm.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.iOS.Runtime.ios-arm.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.iOS.Runtime.ios-arm64.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.iOS.Runtime.ios-arm64.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.iOS.Runtime.ios-x64.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.iOS.Runtime.ios-x64.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.iOS.Runtime.ios-x86.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.iOS.Runtime.ios-x86.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.iOS.Sdk.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.iOS.Sdk.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.macOS.Ref.10.15.100-ci.pr.gh9892.1217+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.macOS.Ref.10.15.100-ci.pr.gh9892.1217+sha.3ce9e3736.nupkg)
* [Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9892.1217+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9892.1217+sha.3ce9e3736.nupkg)
* [Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9892.1217+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9892.1217+sha.3ce9e3736.nupkg)
* [Microsoft.tvOS.Ref.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.tvOS.Ref.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.tvOS.Runtime.tvos-arm64.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.tvOS.Runtime.tvos-arm64.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.tvOS.Runtime.tvos-x64.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.tvOS.Runtime.tvos-x64.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.tvOS.Sdk.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.tvOS.Sdk.14.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.watchOS.Ref.7.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.watchOS.Ref.7.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.watchOS.Runtime.watchos-arm.7.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.watchOS.Runtime.watchos-arm.7.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.watchOS.Runtime.watchos-x86.7.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.watchOS.Runtime.watchos-x86.7.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)
* [Microsoft.watchOS.Sdk.7.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/package/Microsoft.watchOS.Sdk.7.0.100-ci.pr.gh9892.593+sha.3ce9e3736.nupkg)

</details>

✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-9892/3ce9e3736979f3a2534abf137f808113e646f689/1/jenkins-results/generator-diff/index.html) (no change)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.

**Review by spouliot — APPROVED** [review]

**Review by emaf — APPROVED** [review]

File: msbuild/Xamarin.Shared/Xamarin.Shared.props [hunk 1]
```diff
@@ -264,7 +264,8 @@ Copyright (C) 2020 Microsoft. All rights reserved.
 
 	<!-- This is used to determine if VS is connected to mac. -->
 	<PropertyGroup>
-		<IsMacEnabled>true</IsMacEnabled>
+		<IsMacEnabled Condition="'$(IsMacEnabled)' == '' And '$(OS)' != 'Windows_NT'">true</IsMacEnabled>
+		<IsMacEnabled Condition="'$(IsMacEnabled)' == '' And '$(OS)' == 'Windows_NT'">false</IsMacEnabled>
```
Comment (jonathanpeppers) [inline msbuild/Xamarin.Shared/Xamarin.Shared.props:268]: Whoops, I remembered how a user had the `%OS%` environment variable set to `Windows` 🤦 and it breaks this check. It was set on their Dell PC out of the box...

We've been doing this as the new way to check the OS in MSBuild:
* `Condition="$([MSBuild]::IsOSPlatform('windows'))"`
* `Condition="$([MSBuild]::IsOSPlatform('osx'))"`
* `Condition="$([MSBuild]::IsOSPlatform('linux'))"`

Comment (jonathanpeppers) [reply msbuild/Xamarin.Shared/Xamarin.Shared.props:268]: I guess the strings here map to [OSPlatform](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.osplatform?view=netcore-3.1).

https://github.com/dotnet/msbuild/blob/295bf2e8355b6e8e3454a3a3bb4511d255455506/src/Build/Evaluation/IntrinsicFunctions.cs#L439-L442

**Review by whitneyschmidt — APPROVED** [review]

File: msbuild/Xamarin.Shared/Xamarin.Shared.props (hunk 1 above)
Comment (rolfbjarne) [reply msbuild/Xamarin.Shared/Xamarin.Shared.props]: @jonathanpeppers I fixed it to use `$([MSBuild]::IsOSPlatform('windows'))`. Good catch!

**Review by jonathanpeppers — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-9892/2/display/redirect) (on internal Jenkins) succeeded

✅ [Provisioning succeeded](https://jenkins.internalx.com/job/macios/job/PR-9892/2//console)
✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-9892/2//console)
✅ Packages built successfully
<details><summary>View packages</summary>

* [xamarin.ios-14.7.0.95.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/xamarin.ios-14.7.0.95.pkg)
* [Microsoft.iOS.Ref.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.iOS.Ref.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.iOS.Runtime.ios-arm.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.iOS.Runtime.ios-arm.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.iOS.Runtime.ios-arm64.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.iOS.Runtime.ios-arm64.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.iOS.Runtime.ios-x64.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.iOS.Runtime.ios-x64.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.iOS.Runtime.ios-x86.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.iOS.Runtime.ios-x86.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.iOS.Sdk.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.iOS.Sdk.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.macOS.Ref.10.15.100-ci.pr.gh9892.1218+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.macOS.Ref.10.15.100-ci.pr.gh9892.1218+sha.16a9d38ae.nupkg)
* [Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9892.1218+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9892.1218+sha.16a9d38ae.nupkg)
* [Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9892.1218+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9892.1218+sha.16a9d38ae.nupkg)
* [Microsoft.tvOS.Ref.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.tvOS.Ref.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.tvOS.Runtime.tvos-arm64.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.tvOS.Runtime.tvos-arm64.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.tvOS.Runtime.tvos-x64.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.tvOS.Runtime.tvos-x64.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.tvOS.Sdk.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.tvOS.Sdk.14.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.watchOS.Ref.7.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.watchOS.Ref.7.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.watchOS.Runtime.watchos-arm.7.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.watchOS.Runtime.watchos-arm.7.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.watchOS.Runtime.watchos-x86.7.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.watchOS.Runtime.watchos-x86.7.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)
* [Microsoft.watchOS.Sdk.7.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/package/Microsoft.watchOS.Sdk.7.0.100-ci.pr.gh9892.594+sha.16a9d38ae.nupkg)

</details>

✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-9892/16a9d38aee39c66c06b917d5732349648f2db2d1/2/jenkins-results/generator-diff/index.html) (no change)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.

**Review by tj-devel709 — APPROVED** [review]

**Pull request merged**